src/main/java/com/conveyal/r5/analyst/scenario/ShapefileLts.java

```diff
@@ -0,0 +1,84 @@
+package com.conveyal.r5.analyst.scenario;
+
+import com.conveyal.analysis.components.WorkerComponents;
+import com.conveyal.analysis.datasource.DataSourceException;
+import com.conveyal.file.FileStorageFormat;
+import com.conveyal.file.FileStorageKey;
+import com.conveyal.r5.rastercost.ElevationLoader;
+import com.conveyal.r5.shapefile.ShapefileMatcher;
+import com.conveyal.r5.transit.TransportNetwork;
+import com.conveyal.r5.transit.TransportNetworkCache;
+import com.conveyal.r5.util.ExceptionUtils;
+import gnu.trove.list.array.TIntArrayList;
+
+import java.io.File;
+
+import static com.conveyal.file.FileCategory.DATASOURCES;
+
+/**
+ * Custom (experimental) Modification that loads LTS values from a shapefile and matches them to the street edges.
+ * As a custom modification, a model class only exists in R5, there is no separate UI/backend modification class.
+ *
+ * TODO This modification is not effective together with transit - it doesn't cause the egress tables to be rebuilt.
+ * Hmm, actually we don't ever recompute bike egress. Changing the LTS value does not invalidate bike egress tables.
```

Comment:
Does this comment need to be addressed? At the very least, is it clear in the documentation provided to the clients that use this feature?

Reply:
This may be properly handled because affectsStreetLayer() returns true, but I’d like to retrace how that works to make sure it handles this case. Of course additional documentation will be in order if egress is not always affected by modifications.